### ISSUE TITLE
Zettler window stops rendering and window becomes all white - never recovers

### ISSUE THREAD

**Issue by MenooMeow** [Zettlr/Zettlr#5402]
### Description

While clicking onto another .md file to view in the file browser, the Zettler window turned all white and never drew itself.  The only way to use Zettler again was to kill the application and restart it.  It has happened in a few other circumstances but the precise cause is still unknown.  This same issue has occurred other times as well.
![Screenshot_20240929_120122](https://github.com/user-attachments/assets/79477f4d-af34-4ca5-9054-37bec5454dff)

### Reproducing

1. Use the software long enough.  Eventually this will happen.

### Zettlr version

3.2.1.

### Installation Method

From the Website or GitHub

### Installation Method (Other)

DEB download package

### Your Platform

- [ ] Windows
- [ ] macOS
- [X] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit (includes Apple Silicon)

### Operating System Version

Linux Mint 21.3 Cinnamon

**Comment by MenooMeow** [comment]
> > Use the software long enough. Eventually this will happen.
> 
> Can you say something about the exact amount of time one would have to have the application open before this happens? Is it hours? If so, how many?

I would estimate 1 to 2 hrs.  It happened today twice.  What's odd is that sometimes I can leave it going for days without issues.

**Comment by nathanlesage** [comment]
Mh. Could there be something with your system that prompts this issue? I know that we have had this issue in 3.2.0 because of a bug in Electron. Their team is currently somehow changing the window API that has on macOS resulted in blanking out windows. It might be that there is some bug left that affects only Linux systems…?

**Comment by MenooMeow** [comment]
> Mh. Could there be something with your system that prompts this issue? I know that we have had this issue in 3.2.0 because of a bug in Electron. Their team is currently somehow changing the window API that has on macOS resulted in blanking out windows. It might be that there is some bug left that affects only Linux systems…?

I'm doubting it's system related (albeit Linux certainly has tons of hardware issues on laptops) since nothing else is affected.  I'm trying out the latest Zettler update and so far it hasn't happened again.

**Comment by nathanlesage** [comment]
That's a good point, because the previous release also had issues on macOS, as I said. Let's hope for the best! Please close this issue say in a week if nothing happens anymore!

**Comment by MenooMeow** [comment]
Update:  white screen happened again.  This time I left the computer on overnight (as usual).  I locked the screen while I was away for a few hours.  When I came back and restored the Zettler window, it was white.  I "restored" the window from being maximized, dragged it from screen 1 to screen 2 and it drew itself again.

**Comment by MenooMeow** [comment]
Update 2:  it happened again, this time while I was typing a document.  It's a pretty severe problem since I lost all changes made since the last save.

**Comment by Prateek-coder19** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by Carnivorg** [comment]
I have this issue too, using Windows 11 and Zettlr 3.2.3. The window can turn white in as little as 15 minutes sometimes.

I suspect this is related to a memory leak because Task Manager shows Zettlr using up to around 4GB of RAM before this happens.

Fortunately, closing Zettlr from the taskbar opens up the save dialogue, so I haven't lost any data so far...

**Comment by nathanlesage** [comment]
@kyaso CC, I suspect I accidentally created a runaway memory hog. On my system I could observe this too, but I haven't yet found a reliable way to reproduce this.

**Comment by BenCheung0422** [comment]
I am on Windows 11 and I sometimes also face this issue. It happens somehow more easily when I was editing my dozens-KB-sized Markdown file, though quite lag in the first place when editing. The log files do not really describe anything about...perhaps crashing I guess.

**Comment by BabakFarshchian** [comment]
I am not sure if this is helpful, but I had this same issue on Windows 11 on a Thinkpad X1. In my case it seems to have been resolved after I installed some optional driver updates in Windows. They were all Lenovo system updates (firmware) and Logitech driver updates. I am using Zettlr 3.41.

**Comment by garrison** [comment]
I began experiencing this bug a few days ago, right after switching from Silverblue (GNOME) to Kinote (KDE), both based on Fedora 42 and using the same flatpak in both cases.

**Comment by nathanlesage** [comment]
> I began experiencing this bug a few days ago, right after switching from Silverblue (GNOME) to Kinote (KDE), both based on Fedora 42 and using the same flatpak in both cases.

I'm very certain your desktop environment is not the culprit here. Now that you've referenced the other issue, I'm starting to be remembered that I also noticed this once. This seems to be some infinite loop or memory leak, but I haven't yet found any clear causes. Once we have the cause, fixing this is probably going to be very easy (probably, not necessarily…).

**Comment by julesedwards** [comment]
I also encountered the same white screen issue (Zettlr 3.5.1, Windows 11). I too had a very high RAM usage, but it seemed to be related to the use of LangageTool, which once disabled made the RAM usage consistently drop around 300 Mo. My document is quite big and does include custom tables, which was mentionned as being a potential source of the crash, but for the last 3 white screens I had today, I did not modify any table-related stuff...

**Comment by nathanlesage** [comment]
It really sounds as if the main culprit is the table editor. It's about time this thing moved to beta ... @kyaso You're still busy, right? Do you think we're safe moving that to beta and see what breaks...?

**Comment by kyaso** [comment]
@nathanlesage Yeah, unfortunately I'm very busy at the moment, and it won't get better until maybe the end of July 🙈

I'd agree that pushing out a beta is probably the best option here.

**Comment by nathanlesage** [comment]
Thanks for getting back. Then I'll proceed to fast-track the table editor so that we can finally get this off our plates…

**Comment by nathanlesage** [comment]
The TableEditor rewrite is now merged and will be included in tomorrow's nightly. Please keep a look at the performance going forward to see if it really was the tables. If so, we can close this issue. If not, we'll have to go back to searching for the cause.

**Comment by cerquide** [comment]
Seems the bug is back. This is happening to me in Zettlr 4.1.1.
